master/buildbot/process/buildstep.py

```diff
@@ -167,7 +167,9 @@ def _catchup(self):
         def next(x):
             self._catchup()
             return x
-        self.step._start_unhandled_deferreds.append(d)
+
+        if self.step._start_unhandled_deferreds is not None:
```

Comment:
Note that the maybeUnhandled decorator has the same "if..."
https://github.com/buildbot/buildbot/blob/master/master/buildbot/process/buildstep.py#L111